src/test/java/org/minions/devfund/bruno/OceanTest.java

```diff
@@ -0,0 +1,176 @@
+package org.minions.devfund.bruno;
+
+
+import org.junit.Before;
+import org.junit.Test;
+import org.minions.devfund.bruno.battleship.BattleShip;
+import org.minions.devfund.bruno.battleship.Ocean;
+
+import java.util.stream.IntStream;
+
+import static java.util.stream.IntStream.range;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link Ocean}.
+ */
+public class OceanTest {
+
+    private static final int OCEAN_SIZE = 20;
+    private Ocean ocean;
+
+    /**
+     * Setups a new Ocean for the tests.
+     */
+    @Before
+    public void setupOcean() {
+        ocean = new Ocean();
+    }
+
+    /**
+     * Test to Ocean Constructor.
+     */
+    @Test
+    public void testOceanConstructor() {
+        String emptyValueExpected = "empty";
+        int zeroValueExpected = 0;
+        assertEquals(zeroValueExpected, ocean.getHitCount());
+        assertEquals(zeroValueExpected, ocean.getShipsSunk());
+        assertEquals(zeroValueExpected, ocean.getShotsFired());
+        range(0, OCEAN_SIZE).forEach(row ->
+                range(0, OCEAN_SIZE).forEach(col ->
+                        assertEquals(emptyValueExpected, ocean.getShipArray()[row][col].getShipType())));
+    }
+
+    /**
+     * Test place all ships randomly.
+     */
+    @Test
+    public void testPlaceAllShipsRandomly() {
+        final int shipCountExpected = 61;
+        String emptyValueExpected = "empty";
+        int zeroValueExpected = 0;
+        ocean.placeAllShipsRandomly();
+        assertEquals(zeroValueExpected, ocean.getHitCount());
+        assertEquals(zeroValueExpected, ocean.getShipsSunk());
+        assertEquals(zeroValueExpected, ocean.getShotsFired());
+        final int[] shipCountActual = {0};
+        range(0, OCEAN_SIZE).forEach(row ->
+                shipCountActual[0] += Math.toIntExact(range(0, OCEAN_SIZE).filter(col ->
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)